### PR TITLE
Split parallel tests and recycle workers to tame memory usage

### PR DESCRIPTION
**Pull request by sklam**
* Tests are chunked to reduce memory pressure during parallel testing. 
* Seems to speed up TravisCI testing, indicating that our tests are swapping in the build env.

### CODE REVIEW

File: numba/testing/main.py [hunk 1]
```diff
@@ -548,32 +548,39 @@ def _run_inner(self, result):
...
             stests = SerialSuite(self._stests)
             stests.run(result)
             return result
 
-    def _run_parallel_tests(self, result, pool, child_runner):
-        remaining_ids = set(t.id() for t in self._ptests)
-        it = pool.imap_unordered(child_runner, self._ptests)
+    def _run_parallel_tests(self, result, pool, child_runner, tests):
+        remaining_ids = set(t.id() for t in tests)
+        it = pool.imap_unordered(child_runner, tests)
```
Comment (pitrou) [inline numba/testing/main.py]: So, I don't know if that would have the same effect as your approach (I don't understand exactly what's going on here), but `imap_unordered` has an optional `chunksize` argument.

Comment (pitrou) [inline numba/testing/main.py]: Oh, sorry, it becomes clearer if I read your comment above: "recycle the worker process to tame memory usage" :-)

Comment (sklam) [inline numba/testing/main.py]: right, the chunking is really controlling how often the process pool is recycled.

**Comment by sklam** [comment]
This is revealing some access violation on windows.  I'm checking.

**Comment by sklam** [comment]
Ok, it's really coming from MCJIT removeModule leak.  Disabling it fixes the access violations.

**Comment by sklam** [comment]
This patch requires #2294 to run properly on windows.  The access violation can be observed on other platforms but with different test combinations.

**Pull request merged**